### ISSUE TITLE
add support for application/vnd.npm.install-v1+json abbreviated npm install package metadata

### ISSUE THREAD

**Issue by nick-ChenZe** [verdaccio/verdaccio#2792]
**Is your feature request related to a problem?**
Please describe a clear and concise description of what the problem is. E.g. I'm always frustrated when [...]

In our scenario, fetch large size package metadata during `npm install` will crash our private registry.
```
curl -i https://registry.npmjs.org/npm | head -n 10
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/2 200
date: Tue, 21 Dec 2021 09:32:44 GMT
content-type: application/json
content-length: 16813312
```

request an abbreviated document with header `application/vnd.npm.install-v1+json` response smaller

```
curl -i https://registry.npmjs.org/npm -H "Accept: application/vnd.npm.install-v1+json" | head -n 10
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/2 200
date: Tue, 21 Dec 2021 09:33:34 GMT
content-type: application/vnd.npm.install-v1+json
content-length: 1709811
```

https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md#abbreviated-metadata-format

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by juanpicado** [comment]
Great finding I was not aware thanks @nick-ChenZe

**Comment by nick-ChenZe** [comment]
another user case. in node v6.x, npm v3, npm will use metadata to generate `package.json`, instead of use file in `.tar`. In this case, some fields will miss when we set `application/vnd.npm.install-v1+json` to request header. For example

this package request by request with `application/vnd.npm.install-v1+json` has no `main` field, and after execute `npm install` in such version of node and npm, the package in `node_modules` also miss `main` field.
```
{
  "name": "once",
  "versions": {
    "1.1.1": {
      "name": "once",
      "version": "1.1.1",
      "dependencies": {},
      "devDependencies": {
        "tap": "~0.3.0"
      },
      "directories": {
        "test": "test"
      },
      "dist": {
        "shasum": "9db574933ccb08c3a7614d154032c09ea6f339e7",
        "size": 1752,
        "noattachment": false,
        "tarball": "http://npm.registry.baidu-int.com/once/-/once-1.1.1.tgz"
      },
      "_hasShrinkwrap": false,
      "publish_time": 1344929158262,
      "contributors": []
    },
    "1.2.0": {
      "name": "once",
      "version": "1.2.0",
      "dependencies": {},
      "devDependencies": {
        "tap": "~0.3.0"
      },
      "directories": {
        "test": "test"
      },
      "dist": {
        "shasum": "de1905c636af874a8fba862d9aabddd1f920461c",
        "size": 1864,
        "noattachment": false,
        "tarball": "http://npm.registry.baidu-int.com/once/-/once-1.2.0.tgz"
      },
      "_hasShrinkwrap": false,
      "publish_time": 1376276121962,
      "contributors": []
    },
    "1.3.0": {
      "name": "once",
      "version": "1.3.0",
      "dependencies": {},
      "devDependencies": {
        "tap": "~0.3.0"
      },
      "directories": {
        "test": "test"
      },
      "dist": {
        "shasum": "151af86bfc1f08c4b9f07d06ab250ffcbeb56581",
        "size": 1940,
        "noattachment": false,
        "tarball": "http://npm.registry.baidu-int.com/once/-/once-1.3.0.tgz"
      },
      "_hasShrinkwrap": false,
      "publish_time": 1382596046638,
      "contributors": []
    },
    "1.3.2": {
      "name": "once",
      "version": "1.3.2",
      "dependencies": {
        "wrappy": "1"
      },
      "devDependencies": {
        "tap": "~0.3.0"
      },
      "directories": {
        "test": "test"
      },
      "dist": {
        "shasum": "d8feeca93b039ec1dcdee7741c92bdac5e28081b",
        "size": 1776,
        "noattachment": false,
        "tarball": "http://npm.registry.baidu-int.com/once/-/once-1.3.2.tgz"
      },
      "_hasShrinkwrap": false,
      "publish_time": 1430780994026,
      "contributors": []
    },
    "1.3.1": {
      "name": "once",
      "version": "1.3.1",
      "dependencies": {
        "wrappy": "1"
      },
      "devDependencies": {
        "tap": "~0.3.0"
      },
      "directories": {
        "test": "test"
      },
      "dist": {
        "shasum": "f3f3e4da5b7d27b5c732969ee3e67e729457b31f",
        "size": 2000,
        "noattachment": false,
        "tarball": "http://npm.registry.baidu-int.com/once/-/once-1.3.1.tgz"
      },
      "_hasShrinkwrap": false,
      "publish_time": 1411081504887,
      "contributors": []
    },
    "1.3.3": {
      "name": "once",
      "version": "1.3.3",
      "dependencies": {
        "wrappy": "1"
      },
      "devDependencies": {
        "tap": "^1.2.0"
      },
      "directories": {
        "test": "test"
      },
      "dist": {
        "shasum": "b2e261557ce4c314ec8304f3fa82663e4297ca20",
        "size": 1573,
        "noattachment": false,
        "tarball": "http://npm.registry.baidu-int.com/once/-/once-1.3.3.tgz"
      },
      "_hasShrinkwrap": false,
      "publish_time": 1448055914765,
      "contributors": []
    },
    "1.4.0": {
      "name": "once",
      "version": "1.4.0",
      "dependencies": {
        "wrappy": "1"
      },
      "devDependencies": {
        "tap": "^7.0.1"
      },
      "directories": {
        "test": "test"
      },
      "dist": {
        "shasum": "583b1aa775961d4b113ac17d9c50baef9dd76bd1",
        "size": 1979,
        "noattachment": false,
        "tarball": "http://npm.registry.baidu-int.com/once/-/once-1.4.0.tgz"
      },
      "_hasShrinkwrap": false,
      "publish_time": 1473196269367,
      "contributors": []
    }
  },
  "time": {},
  "users": {},
  "dist-tags": {
    "latest": "1.4.0"
  },
  "_rev": "174-0bc44ce331cff443",
  "_id": "once",
  "readme": "",
  "_attachments": {}
}
```

just a use case, thanks for accept this feature.

**Comment by juanpicado** [comment]
I will add this to v6

**Comment by juanpicado** [comment]
Included on `verdaccio/5.15.0`
